MergerLogic/Clients/HeartbeatClient.cs

```diff
@@ -0,0 +1,70 @@
+using MergerLogic.Utils;
+using Microsoft.Extensions.Logging;
+using System.Timers;
+
+namespace MergerLogic.Clients
+{
+    public class HeartbeatClient : IHeartbeatClient
+    {
+        private readonly ILogger _logger;
+        private readonly IConfigurationManager _configurationManager;
+        private readonly IHttpRequestUtils _httpClient;
+        private System.Timers.Timer _timer;
+        private readonly string _baseUrl;
+        private readonly int _intervalMs;
+        private string _taskId;
+
+        public HeartbeatClient(ILogger<GpkgClient> logger, IConfigurationManager configurationManager, IHttpRequestUtils httpClient)
+        {
+            this._logger = logger;
+            this._configurationManager = configurationManager;
+            this._httpClient = httpClient;
+            this._baseUrl = this._configurationManager.GetConfiguration("HEARTBEAT", "baseUrl");
+            this._intervalMs = this._configurationManager.GetConfiguration<int>("HEARTBEAT", "intervalMs");
+        }
+
+        public void Start(string taskId)
+        {
+            if (this._timer != null) {
+                this.Stop();
+            }
+            this._logger.LogInformation($"Starts heartbeats for task={taskId}");
+            this._timer = new System.Timers.Timer();
+            this._timer.Enabled = true;
+            this._taskId = taskId;
+            this._timer.Interval = this._intervalMs;
+            this._timer.Elapsed += this.Send;
```

Comment:
Why are you registering this each time? Maybe initialize the timer in the constructor and just use Timer.Stop and Timer.Start?
https://learn.microsoft.com/en-us/dotnet/api/system.timers.timer.start?view=net-6.0
You will still need to call Timer.Dispose or Timer.Close in case of an exception or when the Hertbeat Distructor is activated.

Reply:
fixed